src/main/java/today/sesac/shoutify/post/dto/response/PostSingleQueryResponseDto.java

```diff
@@ -0,0 +1,45 @@
+package today.sesac.shoutify.post.dto.response;
+
+import jakarta.validation.constraints.NotNull;
+import java.time.LocalDateTime;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+/**
+ * 게시글 단건 조회 응답 dto.
+ */
+@Getter
+@AllArgsConstructor(staticName = "of")
+public class PostSingleQueryResponseDto {
+
+    @NotNull
+    private final Long postId;
```

Comment:
P1
근데, 여긴 Long으로 하고, Notnull일 필요는 없을 것 같습니다.
```suggestion
    private final long postId;
```

Reply:
확인해 봤는데,
`localhost:8080/api/v1/posts/`
이 경로로 하면, 아예 매핑이 안 되네요. 그니까 Long으로 받아도 pathvariable이라서 null 허용이 안 되는 것 같다는 의미입니다~
Controller까지 오지도 못하고 멈추네요.. 신기...
```
org.springframework.web.servlet.NoHandlerFoundException: No endpoint GET /api/v1/posts/.
```
디버그 포인트 찍어 보실 거면, 에러로그에서 보는 것처럼 이 위치에 들어가면 controller로 안 넘어가고, 여기서 에러로 넘어가는 걸 확인할 수 있긴 한데, 로직은... 10분 넘게 뒤지다가.. .어디서 파싱하는지 도무지 모르겠어서 관뒀습니다... ㅋㅋㅋㅋㅎㅎ
 
```
  at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:979) ~[spring-webmvc-6.2.7.jar:6.2.7]
```

Reply:
추가로... @qkralstjr 님이 물어보셔서 다시 찾아 봤는데, response에는 notnull 등 어노테이션을 달아도 동작이 안 됩니다~! 왜냐면... @Valid 어노테이션이 없기 때문임다.

사실 그래서...... 제가 놓쳤는데..... command controller에서 dto를 파라미터로 받을 때는 파라미터에 @valid 어노테이션도 같이 적어주셔야 해요. 지금 아마 테스트해 보시면 null로 보내도 다 통과될 거예요... ㅎㅎ....... 

무튼 그래서 하고자 하는 말은... command controller에 나중에 dto 받는 곳에 @valid를 적어야 한다는 것.
그리고, 모든 response에는 어노테이션을 달아도 동작하는 곳이 없다는 것... (@validated 인가를 달면 된다는 것도 있는데, 아마 이것도 파라미터에서 받을 때 이렇게 사용할 수 있다는 것 같습니다. 왜냐면 @valid 로 검증해 주는 jakarta.validation이 컨트롤러 단에서 검증해 주거든요.

[참고](https://yulee.tistory.com/115) - 공식 문서는 못 찾았는데, 여기에서 이 부분 포함 참고해 주세요~.
> 유효성을 체크할 객체를 "검사하겠다"는 의미로 @Valid를 붙여준다.
모든 요청은 프론트 컨트롤러인 디스패처 서블릿을 통해 컨트롤러로 전달되는데 전달 과정에서 컨트롤러 메소드의 객체를 만들어주는 ArgumentResolver가 동작한다. @Valid 역시 ArgumentResolver에 의해 처리된다. 따라서 @Valid는 기본적으로 컨트롤러에서만 동작하며 다른 계층에서는 검증이 안된다. 다른 계층에서 유효성 검사를 하기 위해서는 @Validated를 사용해야 한다. 
출처: https://yulee.tistory.com/115 [yulee:티스토리]

Reply:
bean validation 어노테이션들은 기본적으로 요청 데이터만 검증을 하는군요!!! 서버 -> 클라이언트로 나가는 값이라서 따로 검사를 안한다고 합니다! 그런데 스웨거 문서화 할 때는 사용이 가능하다고 합니다 !!
본 댓글 바탕으로 수정하겠습니다